experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,241 @@
+# I would like to credit Dario Mathiä who produced an initial version of the following code
+
+export wreath_macdonald_polynomials, wreath_macdonald_polynomial
+
+# Tools
+
+# Computes the b-invariant
+function b1(lambda::Vector{Int})
+  return sum((i-1)*l_i for (i, l_i) in enumerate(lambda); init=0)
+end
+
+# Computes the b-invariant of a multipartition
+function b_inv(lbb::Multipartition)
+  return b1(map(sum,lbb)) + length(lbb)*sum(lambda -> b1(data(lambda)),lbb)
+end
+
+function beta_number_to_partition(beta::Vector{Int})
+  lb=Int[]
+  for j in 1:length(beta)
+    nbholes=beta[j]-beta[1]-(j-1)
+    if nbholes >= 1
+      pushfirst!(lb, nbholes)
+    end
+  end
+  return partition(lb)
+end
+
+# Computes the core associated to the coroot
+#The charge of the core is equal to the coroot
+#in the canonical basis (e_i) of Z^I.
+function core(coroot::Vector{Int},r::Int)
+  beta=Int[]
+  m=minimum(coroot)
+  M=maximum(coroot)
+  beta=[k*r+(j-1) for k in m:M for j in 1:r if k <= coroot[j]]
+  return beta_number_to_partition(beta)
+end
+
+#Inspired by Sagemath's algorithm to obtain a partition from core and quotient
+#Be careful: it is not the usual quotient but the one defined in [Gordon 2008]
+function tau_om(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int})
+   r=length(lbb)
+   w0=perm([r-i for i in 0:(r-1)])  #cf Remark 3.5 Orr and Shimozono
+   lbb_perm=multipartition(permuted(lbb.mp,wperm*w0))
+   gamma=core(coroot,r)
+   lg=length(gamma)
+   k=r*maximum(length(lbb_perm[i]) for i in 1:r) + lg
+   v=[[gamma[i]-(i-1) for i in 1:lg]; [-i for i in lg:(k-1)]]
+   w=[[x for x in v if mod((x-i),r) == 0] for i in 1:r]
+   new_w=Int[]
+   for i in 1:r
+     lw=length(w[i])
+     lq=length(lbb_perm[i])
+     append!(new_w, w[i][1:lq] + r*lbb_perm[i])
+     append!(new_w, w[i][lq+1:lw])
+   end
+   sort!(new_w,rev=true)
+   new_w=[new_w[i]+(i-1) for i in 1:length(new_w)]
+   filter!(x-> x!=0,new_w)
+   return partition(new_w)
+end
+
+#Reoders the CharTable from Chevie
+function reorder(charTirr::Matrix{QQAbFieldElem{AbsSimpleNumFieldElem}}, Modules::Vector{Multipartition{Int}},  mps::Vector{Multipartition{Int}})
```

Comment:
No need to be overly specific with the type here (this way your code is more flexible for future changes)
```suggestion
function reorder(charTirr::Matrix, Modules::Vector{Multipartition{Int}},  mps::Vector{Multipartition{Int}})
```